File: server/withssafy/src/main/java/com/ssafy/withssafy/entity/Comment.java
package com.ssafy.withssafy.entity;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Set;

import javax.persistence.*;

@Entity
@Table(name = "tbl_comment")
@Getter

@NoArgsConstructor
public class Comment {
    @Id @GeneratedValue
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "board_id")
    private Board board;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Comment parent;

    @Column
    private String content;

    @Column
    private String write_dt;

    public void setParentId(Long id){
        if(id != null)
            parent = Comment.builder().id(id).build();
    }

    @Builder
    public Comment(Long id, Board board, User user, Comment parent, String content, String write_dt){
        this.id = id;
        this.board = board;
        this.user = user;
        this.parent = parent;
        this.content = content;
        this.write_dt = write_dt;
    }
}

File: application/WithSSAFY/app/src/main/java/com/ssafy/withssafy/src/dto/board/Comment.kt
package com.ssafy.withssafy.src.dto.board

import com.ssafy.withssafy.src.dto.study.StudyBoard
import com.ssafy.withssafy.src.dto.study.UserX

data class Comment(
    val id: Int,
    val parentId: Int?,
    val boardId: Int,
    val userId: Int,
    val content: String,
    val write_dt: String?,
    val studyBoard: StudyBoard?,
    val user: UserX?
) {
    constructor(boardId: Int, userId: Int, content: String) : this(0, 0, boardId, userId, content, null,null,null)  // 댓글 입력
    constructor(parent: Int, boardId: Int, userId: Int, content: String) : this(0, parent, boardId, userId, content, null,null,null)  // 대댓글 입력
    constructor(id: Int, content: String) : this(id, 0, 0, 0, content, null,null,null)
}